app/assets/javascripts/admin/views/broken-links-report.js

```diff
@@ -0,0 +1,60 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function BrokenLinksReport (module) {
+    this.module = module
+  }
+
+  BrokenLinksReport.prototype.init = function () {
+    if (!fetch) return
+    var button = this.module.querySelector('button')
+    var authenticityTokenField = this.module.querySelector('[name="authenticity_token"]')
+    if (button && authenticityTokenField) this.setupEventListener(button, authenticityTokenField.value)
+    var refreshLink = this.module.querySelector('.js-broken-links-refresh')
+    if (refreshLink) {
+      this.setupPolling(refreshLink.href)
+      refreshLink.remove()
+    }
+  }
+
+  BrokenLinksReport.prototype.setupEventListener = function (button, authenticityToken) {
+    var action = this.module.querySelector('form').action
+    button.addEventListener('click', function (event) {
+      fetch(action + '.json', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ authenticity_token: authenticityToken })
+      }).then(function (response) { return response.json() })
+        .catch(function () { location.reload() })
+        .then(function (json) {
+          this.replaceContentsAndCopyClasses(json.html)
+          this.init()
+        }.bind(this))
+      event.preventDefault()
```

Comment:
Generally, if `event.preventDefault()` is used, it's usually added to the first line of the event function.

```js
button.addEventListener('click', function (event) {
      event.preventDefault()
      ....
})

```